services.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env zsh
+
+# Script used to generate the file ./src/networking/services.rs
+# The source for port-to-service mappings is provided by nmap
+# Available at: https://raw.githubusercontent.com/nmap/nmap/master/nmap-services
+
+OUT=./src/networking/services.rs
+
+{
+  printf '//! This file is generated automatically via ./services.sh\n\n'
+  printf 'use crate::networking::types::protocol::Protocol;\n\n'
+  printf '#[allow(clippy::too_many_lines, clippy::unnested_or_patterns)]\n'
+  printf 'pub fn get_service(port_info: (u16, Protocol)) -> String {\n'
```

Comment:
```suggestion
  printf 'pub fn get_service(port_info: (u16, Protocol)) -> Option<&'static str> {\n'
```

so the caller can decide how they want to represent unknown services and whether they need owned String.  If you care about performance you should avoid String allocations.